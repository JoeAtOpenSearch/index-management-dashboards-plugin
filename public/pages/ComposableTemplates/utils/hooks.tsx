/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { CoreStart } from "opensearch-dashboards/public";
import { CommonService } from "../../../services";
import { TemplateItem, TemplateItemRemote } from "../../../../models/interfaces";

export const getTemplate = async (props: {
  templateName: string;
  commonService: CommonService;
  coreService: CoreStart;
}): Promise<TemplateItem> => {
  const response = await props.commonService.apiCaller<{
    index_templates: { name: string; index_template: TemplateItemRemote }[];
  }>({
    endpoint: "transport.request",
    data: {
      method: "GET",
      path: `_index_template/${props.templateName}?flat_settings=true`,
    },
  });
  let error: string = "";
  if (response.ok) {
    const findItem = response.response?.index_templates?.find((item) => item.name === props.templateName);
    if (findItem) {
      const templateDetail = findItem.index_template;
      return JSON.parse(JSON.stringify(templateDetail));
    }
    error = `The template [${props.templateName}] does not exist.`;
  } else {
    error = response.error || "";
  }

  props.coreService.notifications.toasts.addDanger(error);
  throw new Error(error);
};
