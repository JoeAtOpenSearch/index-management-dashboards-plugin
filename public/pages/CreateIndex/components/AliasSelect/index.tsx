/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { forwardRef } from "react";
import { EuiComboBoxProps } from "@elastic/eui";
import RemoteSelect, { RemoteSelectProps } from "../../../../components/RemoteSelect";
import { ServerResponse } from "../../../../../server/models/types";

export interface AliasSelectProps extends Omit<EuiComboBoxProps<{ label: string; value: string }>, "value" | "onChange"> {
  value?: Record<string, {}>;
  onChange?: (value: AliasSelectProps["value"]) => void;
  refreshOptions: (aliasName: string) => Promise<ServerResponse<{ alias: string }[]>>;
}

const transformObjToArray = (obj: AliasSelectProps["value"]): { label: string }[] => {
  return Object.keys(obj || {}).map((label) => ({ label }));
};
const transformArrayToObj = (array: { label: string }[]): AliasSelectProps["value"] => {
  return array.reduce((total, current) => ({ ...total, [current.label]: {} }), {});
};

const AliasSelect = forwardRef((props: AliasSelectProps, ref: React.Ref<HTMLInputElement>) => {
  const { value, onChange, refreshOptions: refreshOptionsFromProps } = props;
  const refreshOptions: RemoteSelectProps["refreshOptions"] = ({ searchValue }) => {
    return refreshOptionsFromProps(searchValue || "").then((res) => {
      if (res.ok) {
        return {
          ...res,
          response: [...new Set(res.response.map((item) => item.alias).filter((item) => !item.startsWith(".")))].map((item) => ({
            label: item,
          })),
        };
      } else {
        return res;
      }
    });
  };
  return (
    <RemoteSelect
      isDisabled={props.isDisabled}
      placeholder="Select or create aliases"
      customOptionText="Add {searchValue} as a new alias"
      refreshOptions={refreshOptions}
      value={transformObjToArray(value).map((item) => item.label)}
      onChange={(val) => {
        onChange && onChange(transformArrayToObj(val.map((label) => ({ label }))));
      }}
    />
  );
});

AliasSelect.displayName = "AliasSelect";

// @ts-ignore
export default AliasSelect;
