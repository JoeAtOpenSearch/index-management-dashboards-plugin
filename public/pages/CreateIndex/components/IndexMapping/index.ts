/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexMapping from "./IndexMapping";

export default IndexMapping;
