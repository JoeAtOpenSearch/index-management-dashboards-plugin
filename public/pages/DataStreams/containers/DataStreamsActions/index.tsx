/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React, { useMemo, useState, useContext, useCallback } from "react";
import { RouteComponentProps } from "react-router-dom";
import { EuiButton, EuiContextMenu } from "@elastic/eui";
import SimplePopover from "../../../../components/SimplePopover";
import FlushIndexModal from "../../../../containers/FlushIndexModal";
import DeleteIndexModal from "../DeleteDataStreamsModal";
import { ROUTES } from "../../../../utils/constants";
import { DataStream } from "../../../../../server/models/interfaces";
import { ServicesContext } from "../../../../services";
import { BrowserServices } from "../../../../models/interfaces";
import { dataStreamBlockedPredicate, filterBlockedItems } from "../../../../utils/helpers";
import { IndexOpBlocksType } from "../../../../utils/constants";

export interface DataStreamsActionsProps {
  selectedItems: DataStream[];
  onDelete: () => void;
  history: RouteComponentProps["history"];
}

export default function DataStreamsActions(props: DataStreamsActionsProps) {
  const { selectedItems, onDelete, history } = props;
  const [deleteIndexModalVisible, setDeleteIndexModalVisible] = useState(false);
  const [flushDataStreamModalVisible, setFlushDataStreamModalVisible] = useState(false);
  const [blockedDataStreams, setBlockedDataStreams] = useState<string[]>([]);
  const [flushableDataStreams, setFlushableDataStreams] = useState<string[]>([]);
  const services = useContext(ServicesContext) as BrowserServices;

  const onDeleteIndexModalClose = () => {
    setDeleteIndexModalVisible(false);
  };

  const onFlushDataStreamModalClose = () => {
    setFlushDataStreamModalVisible(false);
  };

  const onFlushModalClick = useCallback(async () => {
    const result = await filterBlockedItems<DataStream>(services, selectedItems, IndexOpBlocksType.Closed, dataStreamBlockedPredicate);
    setFlushableDataStreams(result.unBlockedItems.map((item) => item.name));
    setBlockedDataStreams(result.blockedItems.map((item) => item.name));
    setFlushDataStreamModalVisible(true);
  }, [selectedItems, services]);

  const renderKey = useMemo(() => Date.now(), [selectedItems]);
  const selectedItemsInString = selectedItems.map((item) => item.name);

  return (
    <>
      <SimplePopover
        data-test-subj="moreAction"
        panelPaddingSize="none"
        button={
          <EuiButton iconType="arrowDown" iconSide="right">
            Actions
          </EuiButton>
        }
      >
        <EuiContextMenu
          initialPanelId={0}
          // The EuiContextMenu has bug when testing in jest
          // the props change won't make it rerender
          key={renderKey}
          panels={[
            {
              id: 0,
              items: [
                {
                  name: "Force merge",
                  "data-test-subj": "ForceMergeAction",
                  onClick: () => {
                    props.history.push(`${ROUTES.FORCE_MERGE}/${selectedItemsInString.join(",")}`);
                  },
                },
                {
                  name: "Roll over",
                  disabled: selectedItems.length > 1,
                  "data-test-subj": "rolloverAction",
                  onClick: () => history.push(`${ROUTES.ROLLOVER}/${selectedItemsInString.join(",")}`),
                },
                {
                  name: "Flush",
                  disabled: !selectedItems.length,
                  "data-test-subj": "Flush Action",
                  onClick: () => onFlushModalClick,
                },
                {
                  name: "Delete",
                  disabled: selectedItems.length < 1,
                  "data-test-subj": "deleteAction",
                  onClick: () => setDeleteIndexModalVisible(true),
                },
              ],
            },
          ]}
        />
      </SimplePopover>
      <DeleteIndexModal
        selectedItems={selectedItemsInString}
        visible={deleteIndexModalVisible}
        onClose={onDeleteIndexModalClose}
        onDelete={() => {
          onDeleteIndexModalClose();
          onDelete();
        }}
      />

      <FlushIndexModal
        visible={flushDataStreamModalVisible}
        onClose={onFlushDataStreamModalClose}
        flushTarget="data stream"
        flushableItems={flushableDataStreams}
        blockedItems={blockedDataStreams}
      />
    </>
  );
}
