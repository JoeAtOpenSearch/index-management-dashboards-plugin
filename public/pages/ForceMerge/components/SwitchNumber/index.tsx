import React, { useState } from "react";
import { EuiRadioGroup, EuiSpacer } from "@elastic/eui";
import CustomFormRow from "../../../../components/CustomFormRow";
import { AllBuiltInComponents } from "../../../../components/FormGenerator";

interface SwitchNumberProps {
  value?: number;
  onChange: (val: SwitchNumberProps["value"]) => void;
}

export default function SwitchNumber(props: SwitchNumberProps) {
  const [id, setId] = useState(props.value && props.value > 0 ? "1" : "0");
  return (
    <>
      <EuiRadioGroup
        options={[
          {
            id: "0",
            label: "Automatically",
          },
          {
            id: "1",
            label: "Manually set",
          },
        ]}
        idSelected={id}
        onChange={(id) => {
          setId(id);
          if (id === "0") {
            props.onChange(undefined);
          } else {
            props.onChange(1);
          }
        }}
      />
      {id === "1" ? (
        <>
          <EuiSpacer size="s" />
          <CustomFormRow helpText="Specify the number of segments to merge to">
            <AllBuiltInComponents.Number {...props} />
          </CustomFormRow>
        </>
      ) : null}
    </>
  );
}
