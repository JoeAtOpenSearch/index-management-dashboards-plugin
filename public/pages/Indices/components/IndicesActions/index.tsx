/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React, { useCallback, useContext, useMemo, useState } from "react";
import {
  // @ts-ignore
  Criteria,
  // @ts-ignore
  Pagination,
  EuiButton,
  EuiContextMenu,
} from "@elastic/eui";

import { ManagedCatIndex } from "../../../../../server/models/interfaces";
import { ServerResponse } from "../../../../../server/models/types";
import ApplyPolicyModal from "../../components/ApplyPolicyModal";
import SimplePopover from "../../../../components/SimplePopover";
import { ModalConsumer } from "../../../../components/Modal";
import { CoreServicesContext } from "../../../../components/core_services";
import DeleteIndexModal from "../../components/DeleteIndexModal";
import { ServicesContext } from "../../../../services";

export interface IndicesActionsProps {
  selectedItems: ManagedCatIndex[];
  onDelete: (indices: string) => Promise<ServerResponse<any>>;
}

export default function IndicesActions(props: IndicesActionsProps) {
  const { selectedItems, onDelete } = props;
  const [deleteIndexModalVisible, setDeleteIndexModalVisible] = useState(false);
  const context = useContext(CoreServicesContext);
  const services = useContext(ServicesContext);

  const onDeleteIndexModalClose = () => {
    setDeleteIndexModalVisible(false);
  };

  const onDeleteIndexModalConfirm = useCallback(async () => {
    const result = await onDelete(selectedItems.map((item) => item.index).join(","));
    if (result && result.ok) {
      onDeleteIndexModalClose();
    }
  }, [selectedItems, onDelete, services, context, onDeleteIndexModalClose]);

  const renderKey = useMemo(() => Date.now(), [selectedItems]);

  return (
    <>
      <ModalConsumer>
        {({ onShow }) => (
          <SimplePopover
            data-test-subj="More Action"
            panelPaddingSize="none"
            button={
              <EuiButton iconType="arrowDown" iconSide="right">
                Actions
              </EuiButton>
            }
          >
            <EuiContextMenu
              initialPanelId={0}
              // The EuiContextMenu has bug when testing in jest
              // the props change won't make it rerender
              key={renderKey}
              panels={[
                {
                  id: 0,
                  items: [
                    {
                      name: "Delete",
                      disabled: !selectedItems.length,
                      "data-test-subj": "Delete Action",
                      onClick: () => setDeleteIndexModalVisible(true),
                    },
                    {
                      name: "Apply policy",
                      disabled: !selectedItems.length,
                      "data-test-subj": "Apply policyButton",
                      onClick: () =>
                        onShow(ApplyPolicyModal, {
                          indices: selectedItems.map((item: ManagedCatIndex) => item.index),
                          core: context,
                        }),
                    },
                  ],
                },
              ]}
            />
          </SimplePopover>
        )}
      </ModalConsumer>
      <DeleteIndexModal
        selectedItems={selectedItems.map((item) => item.index)}
        visible={deleteIndexModalVisible}
        onClose={onDeleteIndexModalClose}
        onConfirm={onDeleteIndexModalConfirm}
      />
    </>
  );
}
