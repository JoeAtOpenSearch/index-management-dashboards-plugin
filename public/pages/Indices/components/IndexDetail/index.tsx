/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useState } from "react";
import { EuiButtonEmpty, EuiCopy } from "@elastic/eui";
import { IndexItem } from "../../../../../models/interfaces";
import { Modal } from "../../../../components/Modal";
import IndicesActions from "../IndicesActions";
import { ManagedCatIndex } from "../../../../../server/models/interfaces";

export interface IndexDetailModalProps {
  index: string;
  onDelete?: () => void;
  record: ManagedCatIndex;
}

export default function IndexDetail(props: IndexDetailModalProps) {
  // const { detail, visible } = props;
  const { index, record, onDelete } = props;
  const [visible, setVisible] = useState(false);
  const [detail, setDetail] = useState({} as IndexItem);
  return (
    <>
      <EuiCopy textToCopy={index}>
        {(copy) => <EuiButtonEmpty size="xs" flush="right" iconType="copyClipboard" onClick={copy} color="text"></EuiButtonEmpty>}
      </EuiCopy>
      <EuiButtonEmpty onClick={() => setVisible(true)} data-test-subj={`view-index-detail-button-${index}`}>
        {index}
      </EuiButtonEmpty>
      <Modal.SimpleModal
        visible={visible}
        maxWidth={false}
        style={{
          width: "60vw",
        }}
        onClose={() => setVisible(false)}
        title={
          <div style={{ display: "flex", justifyContent: "space-between" }}>
            <span>{index}</span>
            <IndicesActions selectedItems={[record]} onDelete={onDelete} />
          </div>
        }
      />
    </>
  );
}
