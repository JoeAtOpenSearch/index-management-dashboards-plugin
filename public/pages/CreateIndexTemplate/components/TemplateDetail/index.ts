/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexDetail from "./TemplateDetail";

export default IndexDetail;
export * from "./TemplateDetail";
