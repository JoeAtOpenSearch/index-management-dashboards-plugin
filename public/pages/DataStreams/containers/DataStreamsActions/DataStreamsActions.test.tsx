/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { browserServicesMock, coreServicesMock } from "../../../../../test/mocks";
import DataStreamsActions, { DataStreamsActionsProps } from "./index";
import { ServicesContext } from "../../../../services";
import { CoreServicesContext } from "../../../../components/core_services";
import { Route, HashRouter as Router, Switch, Redirect } from "react-router-dom";
import { ROUTES } from "../../../../utils/constants";
const historyPushMock = jest.fn();

function renderWithRouter(props: Omit<DataStreamsActionsProps, "history">) {
  return {
    ...render(
      <CoreServicesContext.Provider value={coreServicesMock}>
        <ServicesContext.Provider value={browserServicesMock}>
          <Router>
            <Switch>
              <Route
                path={ROUTES.DATA_STREAMS}
                render={(routeProps) => (
                  <CoreServicesContext.Provider value={coreServicesMock}>
                    <ServicesContext.Provider value={browserServicesMock}>
                      <DataStreamsActions
                        {...props}
                        history={{
                          ...routeProps.history,
                          push: (...args) => {
                            routeProps.history.push(...args);
                            historyPushMock(...args);
                          },
                        }}
                      />
                    </ServicesContext.Provider>
                  </CoreServicesContext.Provider>
                )}
              />
              <Redirect from="/" to={ROUTES.DATA_STREAMS} />
            </Switch>
          </Router>
        </ServicesContext.Provider>
      </CoreServicesContext.Provider>
    ),
  };
}

describe("<DataStreamsActions /> spec", () => {
  it("renders the component and all the actions should be disabled when no items selected", async () => {
    const { container, getByTestId } = renderWithRouter({
      selectedItems: [],
      onDelete: () => null,
    });

    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });

    userEvent.click(document.querySelector('[data-test-subj="moreAction"] button') as Element);
    await waitFor(() => {
      expect(getByTestId("deleteAction")).toBeDisabled();
    });
  });

  it("delete data streams by calling commonService", async () => {
    const onDelete = jest.fn();
    browserServicesMock.commonService.apiCaller = jest.fn(
      async (): Promise<any> => {
        return { ok: true, response: {} };
      }
    );
    const { container, getByTestId, getByPlaceholderText } = renderWithRouter({
      selectedItems: [{ name: "test_data_stream" }],
      onDelete,
    });

    await waitFor(() => {
      expect(container.firstChild).toMatchSnapshot();
    });

    userEvent.click(document.querySelector('[data-test-subj="moreAction"] button') as Element);
    userEvent.click(getByTestId("deleteAction"));
    userEvent.type(getByPlaceholderText("delete"), "delete");
    userEvent.click(getByTestId("deleteConfirmButton"));

    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledTimes(1);
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "transport.request",
        data: {
          path: `/_data_stream/test_data_stream`,
          method: "DELETE",
        },
      });
      expect(onDelete).toHaveBeenCalledTimes(1);
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledTimes(1);
      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledWith("Delete [test_data_stream] successfully");
    });
  }, 30000);

  it("refresh data streams by calling commonService", async () => {
    browserServicesMock.commonService.apiCaller = jest.fn(
      async (payload): Promise<any> => {
        if (payload.endpoint === "cluster.state") {
          return {
            ok: true,
            response: {
              blocks: {
                indices: {
                  ".ds-blocked-000001": {
                    "4": {},
                  },
                },
              },
            },
          };
        } else if (payload.endpoint === "indices.refresh") {
          return {
            ok: true,
            response: {},
          };
        }
      }
    );

    const { getByTestId, getByText } = renderWithRouter({
      selectedItems: [
        {
          name: "unblocked_data_stream",
          indices: [
            {
              index_name: ".ds-unblocked-000001",
            },
            {
              index_name: ".ds-blocked-000002",
            },
          ],
        },
        {
          name: "blocked_data_stream",
          indices: [
            {
              index_name: ".ds-blocked-000001",
            },
            {
              index_name: ".ds-blocked-000002",
            },
          ],
        },
      ],
    });

    userEvent.click(document.querySelector('[data-test-subj="moreAction"] button') as Element);
    userEvent.click(getByTestId("refreshAction"));
    await waitFor(() => {
      getByText("The following datastream will be refreshed.");
      expect(getByTestId("UnblockedItem-unblocked_data_stream")).not.toBeNull();
      getByText("The following datastream will not be refreshed because they are closed.");
      expect(getByTestId("BlockedItem-blocked_data_stream")).not.toBeNull();
    });

    userEvent.click(getByTestId("refreshConfirmButton"));

    await waitFor(() => {
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "cluster.state",
        data: {
          metric: "blocks",
        },
      });
      expect(browserServicesMock.commonService.apiCaller).toHaveBeenCalledWith({
        endpoint: "indices.refresh",
        data: {
          index: "unblocked_data_stream",
        },
      });

      expect(document.body).toMatchSnapshot();

      expect(coreServicesMock.notifications.toasts.addSuccess).toHaveBeenCalledWith(
        "Refresh datastream [unblocked_data_stream] successfully"
      );
    });
  }, 30000);

  it("refresh data streams disabled because all indexes are closed calling commonService", async () => {
    browserServicesMock.commonService.apiCaller = jest.fn(
      async (payload): Promise<any> => {
        if (payload.endpoint === "cluster.state") {
          return {
            ok: true,
            response: {
              blocks: {
                indices: {
                  ".ds-blocked-000001": {
                    "4": {},
                  },
                },
              },
            },
          };
        } else if (payload.endpoint === "indices.refresh") {
          return {
            ok: true,
            response: {},
          };
        }
      }
    );

    const { getByTestId, getByText } = renderWithRouter({
      selectedItems: [
        {
          name: "blocked_data_stream",
          indices: [
            {
              index_name: ".ds-blocked-000001",
            },
          ],
        },
      ],
    });

    userEvent.click(document.querySelector('[data-test-subj="moreAction"] button') as Element);
    userEvent.click(getByTestId("refreshAction"));
    await waitFor(() => {
      getByText("The following datastream will not be refreshed because they are closed.");
      expect(getByTestId("BlockedItem-blocked_data_stream")).not.toBeNull();
      expect(getByTestId("refreshConfirmButton")).toBeDisabled();
    });
  }, 30000);

  it("refresh data streams even failed to get index status", async () => {
    browserServicesMock.commonService.apiCaller = jest.fn(
      async (payload): Promise<any> => {
        if (payload.endpoint === "cluster.state") {
          throw "failed to call cluster.state";
        } else if (payload.endpoint === "indices.refresh") {
          return {
            ok: true,
            response: {},
          };
        }
      }
    );

    const { getByTestId, getByText } = renderWithRouter({
      selectedItems: [
        {
          name: "blocked_data_stream",
          indices: [
            {
              index_name: ".ds-blocked-000001",
            },
          ],
        },
      ],
    });

    userEvent.click(document.querySelector('[data-test-subj="moreAction"] button') as Element);
    userEvent.click(getByTestId("refreshAction"));
    await waitFor(() => {
      getByText("The following datastream will be refreshed.");
      expect(getByTestId("UnblockedItem-blocked_data_stream")).not.toBeNull();
      expect(getByTestId("refreshConfirmButton")).toBeEnabled();
    });
  }, 30000);
});
