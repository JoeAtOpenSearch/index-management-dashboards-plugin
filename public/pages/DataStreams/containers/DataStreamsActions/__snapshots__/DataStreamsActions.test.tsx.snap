// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataStreamsActions /> spec delete data streams by calling commonService 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;

exports[`<DataStreamsActions /> spec refresh data streams by calling commonService 1`] = `
<body
  class="euiBody-hasOverlayMask"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      class="euiPopover euiPopover--anchorDownCenter euiPopover-isOpen"
      data-test-subj="moreAction"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButton euiButton--primary"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            EuiIconMock
            <span
              class="euiButton__text"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="euiOverlayMask euiOverlayMask--aboveHeader"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="euiModal euiModal--maxWidth-default"
        tabindex="0"
      >
        <button
          aria-label="Closes this modal window"
          class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
          type="button"
        >
          EuiIconMock
        </button>
        <div
          class="euiModal__flex"
        >
          <div
            class="euiModalHeader"
          >
            <div
              class="euiModalHeader__title"
            >
              Refresh 
              data streams
            </div>
          </div>
          <div
            class="euiModalBody"
          >
            <div
              class="euiModalBody__overflow"
            >
              <div
                style="line-height: 1.5;"
              >
                <p>
                  The following 
                  data streams
                   will be refreshed.
                </p>
                <ul
                  style="list-style-type: disc; list-style-position: inside;"
                >
                  <li
                    data-test-subj="UnblockedItem-unblocked_data_stream"
                  >
                    unblocked_data_stream
                  </li>
                </ul>
                <div
                  class="euiSpacer euiSpacer--l"
                />
                <div
                  class="euiCallOut euiCallOut--warning"
                >
                  <div
                    class="euiCallOutHeader"
                  >
                    EuiIconMock
                    <span
                      class="euiCallOutHeader__title"
                    >
                      The following data streams will not be refreshed because they are closed.
                    </span>
                  </div>
                </div>
                <p />
                <ul
                  style="list-style-type: disc; list-style-position: inside;"
                >
                  <li
                    data-test-subj="BlockedItem-blocked_data_stream"
                  >
                    blocked_data_stream
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="euiModalFooter"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--primary"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonEmpty__content"
              >
                <span
                  class="euiButtonEmpty__text"
                >
                  Cancel
                </span>
              </span>
            </button>
            <button
              class="euiButton euiButton--primary euiButton--fill"
              data-test-subj="refreshConfirmButton"
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Refresh
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<DataStreamsActions /> spec refresh data streams disabled because all indexes are closed calling commonService 1`] = `
<body
  class="euiBody-hasOverlayMask"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      class="euiPopover euiPopover--anchorDownCenter euiPopover-isOpen"
      data-test-subj="moreAction"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButton euiButton--primary"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            EuiIconMock
            <span
              class="euiButton__text"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="euiOverlayMask euiOverlayMask--aboveHeader"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="euiModal euiModal--maxWidth-default"
        tabindex="0"
      >
        <button
          aria-label="Closes this modal window"
          class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
          type="button"
        >
          EuiIconMock
        </button>
        <div
          class="euiModal__flex"
        >
          <div
            class="euiModalHeader"
          >
            <div
              class="euiModalHeader__title"
            >
              Refresh 
              data streams
            </div>
          </div>
          <div
            class="euiModalBody"
          >
            <div
              class="euiModalBody__overflow"
            >
              <div
                style="line-height: 1.5;"
              >
                <div
                  class="euiSpacer euiSpacer--l"
                />
                <div
                  class="euiCallOut euiCallOut--warning"
                >
                  <div
                    class="euiCallOutHeader"
                  >
                    EuiIconMock
                    <span
                      class="euiCallOutHeader__title"
                    >
                      The following data streams will not be refreshed because they are closed.
                    </span>
                  </div>
                </div>
                <p />
                <ul
                  style="list-style-type: disc; list-style-position: inside;"
                >
                  <li
                    data-test-subj="BlockedItem-blocked_data_stream"
                  >
                    blocked_data_stream
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <div
            class="euiModalFooter"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--primary"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonEmpty__content"
              >
                <span
                  class="euiButtonEmpty__text"
                >
                  Cancel
                </span>
              </span>
            </button>
            <button
              class="euiButton euiButton--primary euiButton--fill euiButton-isDisabled"
              data-test-subj="refreshConfirmButton"
              disabled=""
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Refresh
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<DataStreamsActions /> spec refresh data streams even failed to get index status 1`] = `
<body
  class="euiBody-hasOverlayMask"
>
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      class="euiPopover euiPopover--anchorDownCenter euiPopover-isOpen"
      data-test-subj="moreAction"
    >
      <div
        class="euiPopover__anchor"
      >
        <button
          class="euiButton euiButton--primary"
          type="button"
        >
          <span
            class="euiButtonContent euiButtonContent--iconRight euiButton__content"
          >
            EuiIconMock
            <span
              class="euiButton__text"
            >
              Actions
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="euiOverlayMask euiOverlayMask--aboveHeader"
  >
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="euiModal euiModal--maxWidth-default"
        tabindex="0"
      >
        <button
          aria-label="Closes this modal window"
          class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
          type="button"
        >
          EuiIconMock
        </button>
        <div
          class="euiModal__flex"
        >
          <div
            class="euiModalHeader"
          >
            <div
              class="euiModalHeader__title"
            >
              Refresh 
              data streams
            </div>
          </div>
          <div
            class="euiModalBody"
          >
            <div
              class="euiModalBody__overflow"
            >
              <div
                style="line-height: 1.5;"
              >
                <p>
                  The following 
                  data streams
                   will be refreshed.
                </p>
                <ul
                  style="list-style-type: disc; list-style-position: inside;"
                >
                  <li
                    data-test-subj="UnblockedItem-blocked_data_stream"
                  >
                    blocked_data_stream
                  </li>
                </ul>
                <div
                  class="euiSpacer euiSpacer--l"
                />
              </div>
            </div>
          </div>
          <div
            class="euiModalFooter"
          >
            <button
              class="euiButtonEmpty euiButtonEmpty--primary"
              type="button"
            >
              <span
                class="euiButtonContent euiButtonEmpty__content"
              >
                <span
                  class="euiButtonEmpty__text"
                >
                  Cancel
                </span>
              </span>
            </button>
            <button
              class="euiButton euiButton--primary euiButton--fill"
              data-test-subj="refreshConfirmButton"
              type="button"
            >
              <span
                class="euiButtonContent euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Refresh
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`<DataStreamsActions /> spec renders the component and all the actions should be disabled when no items selected 1`] = `
<div
  class="euiPopover euiPopover--anchorDownCenter"
  data-test-subj="moreAction"
>
  <div
    class="euiPopover__anchor"
  >
    <button
      class="euiButton euiButton--primary"
      type="button"
    >
      <span
        class="euiButtonContent euiButtonContent--iconRight euiButton__content"
      >
        EuiIconMock
        <span
          class="euiButton__text"
        >
          Actions
        </span>
      </span>
    </button>
  </div>
</div>
`;
